resources/http.go

```diff
@@ -0,0 +1,258 @@
+		},
+		Method: http.MethodGet,
+	}
+}
+
+// Validate if the params passed in are valid data.
+func (obj *HTTPRes) Validate() error {
+	if obj.URL == "" { // this is the only thing that is really required
+		return fmt.Errorf("url can't be empty")
+	}
+
+	return obj.BaseRes.Validate()
+}
+
+// Init runs some startup code for this resource.
+func (obj *HTTPRes) Init() error {
+	obj.client = &http.Client{
+		Timeout: time.Duration(obj.Timeout) * time.Second,
+	}
+
+	obj.SetKind("http")
+	return obj.BaseRes.Init() // call base init, b/c we're overriding
+}
+
+// Watch is the primary listener for this resource and it outputs events.
+func (obj *HTTPRes) Watch() error {
+
+	if err := obj.Running(); err != nil {
+		return err // bubble up a NACK...
+	}
+
+	var send = false // send event?
+	var exit *error
+
+	wg := &sync.WaitGroup{}
+
+	httpResponseChan := make(chan *httpResponse)
+	closeChan := make(chan struct{})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			body := bytes.NewBufferString(obj.Body)
+			req, err := http.NewRequest(obj.Method, obj.URL, body)
+			if err != nil {
+				resp := httpResponse{0, "", err, nil}
+				select {
+				case httpResponseChan <- &resp:
+					return
+				case <-closeChan:
+					return
+				}
+			}
+			req = req.WithContext(ctx)
+
+			res, err := obj.client.Do(req)
+			if err != nil {
+				resp := httpResponse{0, "", err, nil}
+				select {
+				case httpResponseChan <- &resp:
+					return
+				case <-closeChan:
+					return
+				}
+			}
+			response_body, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				resp := httpResponse{res.StatusCode, string(response_body), err, res.Header}
+				select {
+				case httpResponseChan <- &resp:
+					return
+				case <-closeChan:
+					return
+				}
+			}
+			resp := httpResponse{res.StatusCode, string(response_body), nil, res.Header}
+			select {
+			case httpResponseChan <- &resp:
+			case <-closeChan:
+				return
+			}
+		}
+	}()
+	defer wg.Wait()
+	defer close(closeChan)
+	defer cancel()
+	for {
+		select {
+		case event := <-obj.Events():
+			// we avoid sending events on unpause
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+		case result, ok := <-httpResponseChan:
+			if !ok {
+				log.Printf("channel closed")
+				return nil
+			}
+			if result.err != nil {
+				return result.err
+			}
+			obj.responseCode = result.code
+			obj.responseBody = &result.body
+			obj.responseHeader = result.header
+			log.Printf("we got a chan message of %d %s", result.code, result.body)
+			send = true
+			obj.StateOK(false) // dirty
+		}
+		if send {
+			send = false
+			obj.Event()
+		}
+	}
+}
+
+// CheckApply checks the resource state and applies the resource if the bool
+// input is true. It returns error info and if the state check passed or not.
+func (obj *HTTPRes) CheckApply(apply bool) (bool, error) {
+	if val, exists := obj.Recv["URL"]; exists && val.Changed {
```

Comment:
I think you should just take out all of this unused body for now and put // XXX: TODO or something...